src/main/plugins/browser-bookmarks-plugin/firefox-bookmark-repository.ts

```diff
@@ -0,0 +1,69 @@
+        type: IconType.SVG
+    };
+
+    constructor(private readonly userDataFolderPath: string) {}
+
+    public getBrowserBookmarks(): Promise<BrowserBookmark[]> {
+        return this.getDatabaseFilePath()
+            .then(dbFilePath => this.getBookmarks(dbFilePath));
+    }
+
+    private getDatabaseFilePath(): Promise<string> {
+        const profilesIniFilePath = `${this.userDataFolderPath}${path.sep}profiles.ini`;
+        return new Promise((resolve, reject) => FileHelpers.readFile(profilesIniFilePath)
+            .then(fileContent => {
+                const profilesIni = ini.parse(fileContent);
+                const profilePath = this.getAbsoluteProfilePath(profilesIni);
+                if (profilePath) {
+                    resolve(`${profilePath}${path.sep}places.sqlite`);
+                } else {
+                    reject('Profile not found');
+                }
+            })
```

Comment:
Do we need error handling here?

Reply:
I'm not exactly sure what you mean. I would discourage from resolving this promise in case something goes wrong. If someone enables the browse bookmarks plugin and chooses Firefox, an error should be logged if something did not work.

Reply:
Ah yes you're right. I was thinking about adding a `catch` statement here, but it would be unnecessary.